### PR TITLE
Some little cosmetics

### PR DESCRIPTION
**Pull request by carstene1ns**

### CODE REVIEW

File: src/decoder_fmmidi.cpp [hunk 1]
```diff
@@ -57,9 +57,9 @@ bool FmMidiDecoder::Open(FILE* file) {
 	fseek(file, 0, SEEK_END);
 	file_buffer.resize(ftell(file) - old_pos);
 	fseek(file, old_pos, SEEK_SET);
-	fread(file_buffer.data(), file_buffer.size(), 1, file);
+	size_t bytes_read = fread(file_buffer.data(), file_buffer.size(), 1, file);
 
-	if (!seq->load(this, read_func)) {
+	if ((bytes_read =! file_buffer.size()) || (!seq->load(this, read_func))) {
```
Comment (Ghabry) [inline src/decoder_fmmidi.cpp]: =!

Comment (carstene1ns) [inline src/decoder_fmmidi.cpp]: ?

Comment (Ghabry) [inline src/decoder_fmmidi.cpp]: the operator should be != . I guess this assigns 0 or 1 (negation of file_buffer.size() to bytes_read.

File: src/image_png.cpp [hunk 2]
```diff
@@ -187,16 +187,19 @@ static void ReadGrayData(
...
+			uint8_t a = (*p & 0xff000000) >> 24;
+			uint8_t b = (*p & 0x00ff0000) >> 16;
+			uint8_t g = (*p & 0x0000ff00) >> 8;
+			uint8_t r = (*p & 0x000000ff);
+
+			// Black pixels are transparent
+			if (b == 0 && g == 0 && r == 0 && a == 255)
+				*p = 0;
+		}
 	}
```
Comment (Ghabry) [inline src/image_png.cpp]: Finally tested it under the Wii: Does indeed not work. Colors are completely wrong when transparency is used. All other graphics look fine.

**Comment by carstene1ns** [comment]
I expected this. Going to fix this later.

**Comment by Ghabry** [comment]
Do you plan to fix the png reader for the Wii before 0.5.2? Otherwise that commit could be removed and fixed for after 0.5.2 because the current code doesn't break anything.

**Comment by carstene1ns** [comment]
Let's move this after 0.5.2 then. Will do a final rebase in a bit.

**Comment by carstene1ns** [comment]
Please note that there are changes since your review:
- commit 5b52266 is new.
- commit 0c7cbac has been changed significantly in `audio_generic` and `utils`.

File: src/sdl_ui.cpp [hunk 3]
```diff
@@ -786,6 +786,7 @@ void SdlUi::ProcessKeyDownEvent(SDL_Event &evnt) {
 			return;
 		}
 
+		[[gnu::fallthrough]];
```
Comment (fdelapena) [inline src/sdl_ui.cpp]: in GCC6 says:

`warning: attributes at the beginning of statement are ignored [-Wattributes]`

Comment (carstene1ns) [inline src/sdl_ui.cpp]: I had experimented with some "fall-through comments" as mentioned in the link I provided in the commit description, but it did not work.
Also MSVC warns about not recognizing `gnu` attributes (of course^^).
If we do not figure out a better solution, I will move the check for <kbd>Enter</kbd> into the `default` case instead.

Comment (carstene1ns) [inline src/sdl_ui.cpp]: gnulib (mentioned in that bug report) does this:
```c++
#ifndef FALLTHROUGH
# if __GNUC__ < 7
#  define FALLTHROUGH ((void) 0)
# else
#  define FALLTHROUGH __attribute__ ((__fallthrough__))
# endif
#endif
```
and then
```c++
                  case TYPE_ULONGLONGINT:
                    // [...]
                    FALLTHROUGH;
                  case TYPE_LONGINT:
```
m(

**Comment by carstene1ns** [comment]
jenkins: test this please

**Pull request merged**